internal/proto/local/plugin/host_catalog.proto

```diff
@@ -0,0 +1,27 @@
+syntax = "proto3";
+
+package plugin.sdk.proto;
+
+option go_package = "github.com/hashicorp/boundary/internal/plugin/sdk/proto;proto";
+
+import "google/protobuf/struct.proto";
+
+// HostCatalog represents a host catalog as passed in to a plugin call. 
+message HostCatalog {
```

Comment:
Should this include name, description, version, create time, update time, etc?  I'm wondering if this should be as close to the API resource as possible since that is what is what the plugin author will be using as a baseline for understanding what the resource itself is.

Reply:
Actually thinking about this more...  In protos it is easy to add fields as they become necessary but deprecating them can be a little more painful.  I'm ok with keeping the number of fields at a minimum until we identify the need for more fields.

Reply:
Yeah, that was the intention for sure. I was just asking myself the question "are the plugins really going to initially use these fields?", and could not really think of a situation where they would be.

Reply:
Resolving, I think we're on the same page here 🙂

Reply:
`id` also has a meaning in boundary (the boundary supplied resource id) and this field is also included in host catalogs and host sets in this interface so from a plugin perspective they are already exposed to this meaning.  What do you think about `external_id`?

Reply:
Sounds good to me!

Reply:
Hmm, I'll make this change, but wondering if we need something new for plugins to note that this ID may not be exactly what is reflected in Boundary?

Reply:
Woops! I added my comment to the wrong closed conversation.  This was meant for Host and not HostCatalog.  TO be clear, I think Host Catalog and Host Set should use `id` since this is the value provided by boundary.  I think Host should use `external_id` since that is the value that is populated by the plugin and will be used to calculate the stable id.